utils/buttons.py

```diff
@@ -409,41 +458,42 @@ async def wait_for_message(self) -> AsyncGenerator[bool, discord.Message]:
         while True:
             try:
                 message = await self.context.bot.wait_for("message", check=self.predicate, timeout=self.timeout)
-                value = yield message
-                self.reset_timeout()
-                if value:
-                    self.result = message
-                    self.stop()
-                    break
-                if value is False:
-                    error = self.invalid_response()
-                    await message.reply(error, delete_after=60)
-                yield
             except asyncio.TimeoutError:
                 self.stop()
 
-    @ui.button(emoji="<:crossmark:753620331851284480>", label="Cancel", style=discord.ButtonStyle.danger)
-    async def denied_action(self, button: ui.Button, interaction: discord.Interaction):
+            value = yield message
+            self.reset_timeout()
+            if value:
+                self.result = message
+                self.stop()
+                break
+            if value is False:
+                error = self.invalid_response()
+                await message.reply(error, delete_after=60)
+            yield
```

Comment:
Looks like try block was too wide there. TimeoutError only happens in wait_for context